src/Collapse.php

```diff
@@ -4,195 +4,418 @@
+     *
+     * Example usage:
+     * ```php
+     * $collapse->containerAttributes(['data-id' => '123']);
+     * ```
+     */
+    public function containerAttributes(array $containerAttributes): self
     {
-        $options = $this->bodyOptions;
-        Html::addCssClass($options, ['widget' => 'card card-body']);
+        $new = clone $this;
+        $new->containerAttributes = $containerAttributes;
 
-        return $options;
+        return $new;
     }
 
-    protected function prepareToggleOptions(): array
+    /**
+     * Sets the items.
+     *
+     * @param CollapseItem ...$items The items.
+     *
+     * @return self A new instance with the specified items.
+     *
+     * Example usage:
+     * ```php
+     * $collapse->items(
+     *     CollapseItem::widget('Item 1'),
+     *     CollapseItem::widget()->content('Item 2'),
+     * );
+     * ```
+     */
+    public function items(CollapseItem ...$items): self
     {
-        [$tagName, $options, $encode] = parent::prepareToggleOptions();
-
-        $options['aria-expanded'] = $this->collapsed ? 'true' : 'false';
+        $new = clone $this;
+        $new->items = $items;
 
-        return [$tagName, $options, $encode];
+        return $new;
     }
 
-    public function begin(): ?string
+    /**
+     * Sets the HTML attributes for the container of the toggler.
+     *
+     * @param array collapseContainerAttributes Attribute values indexed by attribute names.
+     *
+     * @return self A new instance with the specified attributes for the container of the toggler.
+     *
+     * @see {\Yiisoft\Html\Html::renderTagAttributes()} for details on how attributes are being rendered.
+     *
+     * Example usage:
+     * ```php
+     * $collapse->togglerContainerAttributes(['data-id' => '123']);
+     * ```
+     */
+    public function togglerContainerAttributes(array $togglerContainerAttributes): self
     {
-        parent::begin();
+        $new = clone $this;
+        $new->togglerContainerAttributes = $togglerContainerAttributes;
 
-        $options = $this->prepareOptions();
-        $bodyOptions = $this->prepareBodyOptions();
-        $this->tagName = ArrayHelper::remove($options, 'tag', 'div');
-        $html = $this->renderToggle ? $this->renderToggle() : '';
-        $html .= Html::openTag($this->tagName, $options);
+        return $new;
+    }
 
-        if ($bodyTag = ArrayHelper::remove($bodyOptions, 'tag')) {
-            $html .= Html::openTag($bodyTag, $bodyOptions);
-        }
+    /**
+     * Sets the tag for the container of the toggler.
+     *
+     * @param string $togglerContainerTag The tag for the container of the toggler.
+     *
+     * @return self A new instance with the specified tag for the container of the toggler.
+     *
+     * Example usage:
+     * ```php
+     * $collapse->togglerTag('div');
+     * ```
+     */
+    public function togglerContainerTag(string $togglerContainerTag): self
```

Comment:
Which is correct? Method name or example? Is that container or toggler itself?

Reply:
https://github.com/yiisoft/yii-bootstrap5/blob/89d6f317e062f01e8001ad8c51cea95dd5235781/tests/CollapseTest.php#L456

Reply:
done.